test/metric_value_benchmark/collectd_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+)
+
+type CollectDTestRunner struct {
+}
+
+var _ ITestRunner = (*CollectDTestRunner)(nil)
+
+func (t *CollectDTestRunner) validate() status.TestGroupResult {
+	metricsToFetch := t.getMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		testResults[i] = validateCollectDMetric(metricName)
+	}
+
+	return status.TestGroupResult{
+		Name:        t.getTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (t *CollectDTestRunner) getTestName() string {
+	return "CollectD"
+}
+
+func (t *CollectDTestRunner) getAgentConfigFileName() string {
+	return "collectd_config.json"
+}
+
+func (t *CollectDTestRunner) getAgentRunDuration() time.Duration {
+	return minimumAgentRuntime
+}
+
+func (t *CollectDTestRunner) setupBeforeAgentRun() error {
+	startCollectdCommands := []string{
+		"sudo mkdir -p /etc/collectd",
+		"sudo collectd -C ./extra_configs/collectd.conf",
+	}
+
+	return common.RunCommands(startCollectdCommands)
+}
```

Comment:
The default configuration for `collectd` would includes many other plugins (e.g CPU, Memory, Interface, etc). Therefore, to limit the amount of metrics we set and less rely on the default enable plugins from collectd, it would better to setup the collectd configuration and setup all these steps beforehand.